controllers/pod.go

```diff
@@ -251,6 +252,18 @@ func (r *SingleClusterReconciler) restartPods(
 
 	for idx := range podsToRestart {
 		pod := podsToRestart[idx]
+
+		// Create services for all pods if network policy is changed and rely on nodePort service.
+		// Check podServiceNeeded condition for both status and spec network policy
+		if !podServiceNeeded(r.aeroCluster.Status.PodSpec.MultiPodPerHost, &r.aeroCluster.Status.AerospikeNetworkPolicy) &&
+			podServiceNeeded(r.aeroCluster.Spec.PodSpec.MultiPodPerHost, &r.aeroCluster.Spec.AerospikeNetworkPolicy) {
+			if err := r.createPodService(
+				pod.Name, r.aeroCluster.Namespace,
+			); err != nil && errors.IsAlreadyExists(err) {
```

Comment:
why are we returning error if service is already exist and bypassing other errors?

Reply:
Fixed and moved code block to diff location